models/Article.php

```diff
@@ -106,108 +95,218 @@ function get_post(){
     }
     
     // Fonction permettant de supprimer un article
     function delete_article()
-    {
-        $db = getPdo();
+    {  
+        $id = filter_input(INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT);
         
-        if (empty($_GET['id']) || !ctype_digit($_GET['id'])) {
-            die("Ho ! Fallait préciser le paramètre id en GET !");
-        }
-
-        $id = $_GET['id'];
+        $query = $this->db->prepare('
+        DELETE
+        FROM articles 
+        WHERE id = :id');
+        $query->execute(['id' => $id]
+        );
+    }
+    
+      // Fonction permettant de supprimer un article
+    function delete_article_comments()
+    {  
+        $id = filter_input(INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT);
 
-        $query = $db->prepare('SELECT * FROM articles WHERE id = :id');
+        $query = $this->db->prepare('
+        DELETE 
+        FROM comments 
+        JOIN articles
+        ON comments.article_id = article.id
+        WHERE id = :id;
+        ');
         $query->execute(['id' => $id]);
-        if ($query->rowCount() === 0) {
-            die("Aucun commentaire n'a l'identifiant $id !");
+
+        header("Location: /liste-de-tous-les-articles");
+    }
+    
+    //Fonction permettant de vérifier les données envoyées par l'utilsiateur
+     function form_page_write(){
+        if(isset($_POST['post'])){
+            if(isset($_POST['title'])){
+                $title = htmlspecialchars(trim($_POST['title']));
+            }
+            if(isset($_POST['content'])){
+                $content = htmlspecialchars(trim($_POST['content']));
+            }
+
+            $posted = isset($_POST['public']) ? "1" : "0";
+
+            $errors = [];
+
+            if(empty($title) || empty($content)){
+                $errors['empty'] = "Veuillez remplir tous les champs";
+            }
+
+            if(!empty($_FILES['image']['name'])){
+                $file = $_FILES['image']['name'];
+                $extensions = ['.png','.jpg','.jpeg','.gif','.PNG','.JPG','.JPEG','.GIF'];
+                $extension = strrchr($file,'.');
+                if(!in_array($extension,$extensions)){
+                    $errors['image'] = "Cette image n'est pas valable";
+                }
+            }
+
+            if(!empty($errors)){
+                ?>
+                    <div class="card red">
+                        <div class="card-content white-text">
+                            <?php
+                                foreach($errors as $error){
+                                    echo $error;
+                                }
+                            ?>
+                        </div>
+                    </div>
+                <?php
+            }else{
+                Article::post($title,$content,$posted);
+                if(!empty($_FILES['image']['name'])){
+                    Article::post_img($_FILES['image']['tmp_name'], $extension);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Detected usage of a non-sanitized input variable: $_FILES['image']['tmp_name']](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5344015)